### PR TITLE
docs(core): add docs for parallelism

### PR DESCRIPTION
**Pull request by xiongemi**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: docs/shared/reference/project-configuration.md [hunk 1]
```diff
@@ -200,6 +200,27 @@ If you are using distributed task execution and disable caching for a given targ
 
 {% /callout %}
 
+### Parallelism
+
+In Nx 19.5.0 and higher, you can configure parallelism for a target by setting `"parallelism": false` in the target's configuration. This ensures that the target will not run in parallel with other tasks. For example, if you want the `e2e` target to run tests alone, you can specify:
```
Comment (FrozenPandaz) [inline docs/shared/reference/project-configuration.md:205]:
```suggestion
In Nx 19.5.0+, you can configure parallelism for a target by setting `"parallelism": false` in the target's configuration. This ensures that the target will not run in parallel with other tasks. For example, if you do not want the `e2e` target to alongside any other tasks, you can specify:
```

File: docs/shared/reference/project-configuration.md [hunk 2]
```diff
@@ -200,6 +200,27 @@ If you are using distributed task execution and disable caching for a given targ
...
+{
+  "targets": {
+    "e2e": {
+      "parallelism": false
+    }
+  }
+}
+```
+
+Parallelism is a feature in Nx that allows multiple tasks to run concurrently. By default, Nx tries to execute as many tasks in parallel as possible to optimize build and test times. However, there are scenarios where running tasks in parallel is not desirable. For example, tasks that rely on shared resources or have side effects might fail or produce inconsistent results if run concurrently.
```
Comment (FrozenPandaz) [inline docs/shared/reference/project-configuration.md:217]:
```suggestion
Parallelism is a feature in Nx that allows multiple tasks to run in parallel. By default, Nx tries to execute tasks in parallel to optimize build and test times. However, there are scenarios where running tasks in parallel is not desirable. For example, tasks that rely on shared resources or have side effects might fail or produce inconsistent results when run alongside other tasks.
```

File: docs/shared/reference/project-configuration.md [hunk 3]
```diff
@@ -200,6 +200,27 @@ If you are using distributed task execution and disable caching for a given targ
...
+
+To control parallelism, you can set the "parallelism" option in a target's configuration. This option is available starting from version 19.5.0 and can be set to either true or false.
+
+- `"parallelism": true` (default): The task can run in parallel with other tasks.
+- `"parallelism": false`: The task will run alone, without any other tasks running concurrently.
```
Comment (FrozenPandaz) [inline docs/shared/reference/project-configuration.md:222]: Remove

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Comment by FrozenPandaz** [comment]
This is waiting on Nx Cloud changes

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**